server/src/models/database/socobo-user.ts

```diff
@@ -1,16 +1,64 @@
+import { IsEmail, IsNotEmpty, MinLength, ValidateIf } from "class-validator";
 import { ProviderType } from "./../enums/provider-type";
 import { Role } from "./../enums/role";
+import { ValidationGroup } from "./../enums/validation-group";
 
 export class SocoboUser {
+
   public id: number;
+
+  @ValidateIf((o) => o.email === "", {
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @IsNotEmpty({
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @MinLength(5, {
+    groups: [ ValidationGroup.LOGIN ]
+  })
   public username: string;
+
+  @ValidateIf((o) => o.username === "", {
```

Comment:
Okay, so I would not change any logic, I think this should be fine. A clearer error message would be nice but if this is not easily to accomplish, I would stick with the current version. Maybe this wouldn't be a problem at all since we are the only client at the moment and it is also in the readme. Sorry for the long discussion, I was just a bit confused by this. Maybe you can find a way to improve the message. If not lets go ahead and merge this PR.